synapse/handlers/room_member.py

```diff
@@ -450,8 +455,37 @@ def _update_membership(
             prev_events_and_hashes=prev_events_and_hashes,
             content=content,
         )
+        logcontext.run_in_background(
+            self._send_merged_user_invites,
```

Comment:
don't we hit this codepath for all membership changes, not just invites? surely we shouldn't be sending invites when we leave a room?

Reply:
correct, I misinterpreted the code and thought this was only for joins. Will fix.